e2e/credentials-server.e2e.spec.ts

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * This file includes end-to-end testing for the HTTP control interface
+ */
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { spawnSync } from 'child_process';
+
+import { expect, test } from '@playwright/test';
+import { BrowserContext, ElectronApplication, Page, _electron } from 'playwright';
+
+import fetch from 'node-fetch';
+import { createDefaultSettings, playwrightReportAssets } from './utils/TestUtils';
+import paths from '@/utils/paths';
+import { ServerState } from '@/main/commandServer/httpCommandServer';
+import { spawnFile } from '@/utils/childProcess';
+import { findHomeDir } from '@/config/findHomeDir';
+
+function haveDockerCredentialAssistant(): boolean {
+  // Not using the code from `httpCredentialServer.ts` because we can't use async code at top-level here.
+  const dockerConfigPath = path.join(findHomeDir() ?? '', '.docker', 'config.json');
+
+  try {
+    const contents = JSON.parse(fs.readFileSync(dockerConfigPath).toString());
+    const credStore = contents.credsStore;
+
+    if (!credStore) {
+      return false;
+    }
+    const result = spawnSync(`docker-credential-${ credStore }`, { input: 'list', stdio: 'pipe' });
+
+    return !result.error;
+  } catch (err: any) {
+    return false;
+  }
+}
+
+const testWithCreds = haveDockerCredentialAssistant() ? test : test.skip;
+
+test.describe('Credentials server', () => {
```

Comment:
This whole section only runs if a credential store is set up? (That is, there doesn't seem to be tests for the no credential store case.)

Reply:
More easily done as a unit test.

Reply:
Done in another comment here.